src/all/hennojin/src/eu/kanade/tachiyomi/extension/all/hennojin/Hennojin.kt

```diff
@@ -66,46 +74,47 @@ class Hennojin(override val lang: String, suffix: String) : ParsedHttpSource() {
     override fun searchMangaFromElement(element: Element) =
         popularMangaFromElement(element)
 
-    override fun mangaDetailsRequest(manga: SManga) =
-        GET("https://hennojin.com" + manga.url, headers)
-
     override fun mangaDetailsParse(document: Document) =
         SManga.create().apply {
-            description = document.selectFirst(
+            description = document.select(
                 ".manga-subtitle + p + p",
-            )?.html()?.replace("<br> ", "\n")
+            ).joinToString("\n") {
+                it.run {
+                    select(Evaluator.Tag("br")).prepend("\\n")
+                    this.text().replace("\\n", "\n").replace("\n ", "\n")
+                }
+            }.trim()
             genre = document.select(
                 ".tags-list a[href*=/parody/]," +
                     ".tags-list a[href*=/tags/]," +
                     ".tags-list a[href*=/character/]",
-            )?.joinToString { it.text() }
-            artist = document.select(
+            ).joinToString { it.text() }
+            artist = document.selectFirst(
                 ".tags-list a[href*=/artist/]",
-            )?.joinToString { it.text() }
-            author = document.select(
+            )?.text()
+            author = document.selectFirst(
                 ".tags-list a[href*=/group/]",
-            )?.joinToString { it.text() } ?: artist
+            )?.text() ?: artist
             status = SManga.COMPLETED
         }
 
-    override fun fetchChapterList(manga: SManga) =
-        Request.Builder().url(manga.thumbnail_url!!)
-            .head().build().run(client::newCall)
-            .asObservableSuccess().map { res ->
-                SChapter.create().apply {
-                    name = "Chapter"
-                    url = manga.reader
-                    date_upload = res.date
-                    chapter_number = -1f
-                }.let(::listOf)
-            }!!
-
-    override fun pageListRequest(chapter: SChapter) =
-        GET("https://hennojin.com" + chapter.url, headers)
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup(response.body.string())
+        val date = Request.Builder().url(document.selectFirst(".manga-thumbnail > img")!!.attr("src"))
```

Comment:
Thumbnail should be optional.

Doesn't make sense that parsing the date relies on the thumbnail being present.